benchmarks/tpch/justfile

```diff
@@ -0,0 +1,52 @@
+set fallback
+VENV := ".venv"
+VENV_BIN := ".venv/bin"
+
+
+@venv:
+	@python3 -m pip install virtualenv > /dev/null
+	@python3 -m virtualenv {{VENV}} > /dev/null
+
+## Set up virtual environment and install requirements
+@requirements: venv
+	{{VENV_BIN}}/python -m pip install --upgrade pip > /dev/null
+	{{VENV_BIN}}/pip install -r requirements.txt > /dev/null
+	
+clean-tpch-dbgen:
+	make -C tpch-dbgen clean
+
+clean-venv:
+	rm -r .venv
+
+clean-tables:
+	rm -r tables_scale_*
+
+clean: clean-tpch-dbgen clean-venv
+
+gen_scale_factor n: requirements
+	make -C tpch-dbgen all
+	cd tpch-dbgen && ./dbgen -vf -s {{n}} && cd ..
```

Comment:
not sure i follow?

Reply:
Oh this is just shell scripting nit/nonsense (pushd/popd are like CD but they have a stack of your previous directory changes so you don't have to pair the pop case.

`pushd tpch-dbgen; ./dbgen -vfs {{n}}; popd` 

(I don't think the `cd` can fail, because if it did (permissions non-existantce, line 27 would also fail.) 

I presume that just has the same kind of make semantics with each shell line being run in a different bash session?